airbyte-integrations/connectors/source-chargify/source_chargify/source.py

```diff
@@ -0,0 +1,133 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from urllib.parse import parse_qs, urlparse
+
+import requests
+from requests.auth import HTTPBasicAuth
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.models import SyncMode
+
+
+
+
+# Basic full refresh stream
+class ChargifyStream(HttpStream, ABC):
+
+    def __init__(self, subdomain: str, per_page: int, page: str, *args, **kwargs):
+        super().__init__(**kwargs)
+
+        self._subdomain = subdomain
+        self._per_page = per_page
+        self._page = page
```

Comment:
```suggestion
    PER_PAGE = 200
    FIRST_PAGE = 1
    def __init__(self, subdomain: str, per_page: int, page: str, *args, **kwargs):
        super().__init__(**kwargs)
        self._subdomain = subdomain
```
In your current implementation, the first page number and per page are static values defined in `convert_config2stream_args`. For readability, I find it better to set these as class attributes.

Reply:
yup make sense!